### DISCUSSION TITLE
Valetudo 2021.11.0

### DISCUSSION THREAD

**Discussion by Hypfer** [Hypfer/Valetudo#1175] [Release Discussions]
<div align="center">
    <img src="https://raw.githubusercontent.com/Hypfer/Valetudo/master/assets/logo/valetudo_logo_with_name.svg" width="800" alt="valetudo">
    <p align="center"><h2>2021.11.0</h2></p>
</div>

Likely the most important release of the year.

## Recap 0.6.1 to 2021.11.0
Apparently, Valetudo 0.6.1 was released only a bit more than a year ago, but tbh it feels like an eternity.
So much has changed since then. The iteration speed of the project has been insane in the last months.
As we've now finished most if not all of the huge tasks, I expect things to slow down a little.

Here's a quick recap of what happened since 0.6.1:

- Complete core rewrite to easily support multiple vendors and different robots
- A proper REST-API documented with Swagger UI
- Complete MQTT rewrite to support Home Assistant as well as Homie
- Easy map display in Home Assistant via MQTT and the custom lovelace valetudo map card
- Complete UI rewrite based on a proper UI framework
- Support for all relevant features of the robots
- Support for autodiscovery of Valetudo-enabled robots via mDNS and SSDP
- Support for Dreame-made robots
- An [android companion app](https://valetudo.cloud/pages/companion_apps/valetudo_companion.html) that makes use of said autodiscovery feature
- [Valeronoi](https://valetudo.cloud/pages/companion_apps/valeronoi.html): A companion app that generates Wi-Fi signal strength maps using Valetudo
- An updater functionality
- [Proper docs](https://valetudo.cloud/)
- General polishing. A lot of it

## Default new UI

Starting with this release, the new react-based UI introduced by @jomik with large contributions by @ccoors is now the default.

The existing map renderer was ported to react with a few quality of life improvements:

- You can now specify an iteration count when cleaning segments or zones
- Layers are now sorted, meaning that walls will always be on top. This solves some visual issues of vSlam-based devices
- Adding a new zone/virtual wall/etc. will now spawn it in the middle of your viewport 
- Map segment labels now attempt to actually be inside the segment they're for. This fixes issues with cornered hallways. No more stray triangles
- Editing the map will now refresh your current view instead of navigating back
- Rendering has been partly moved to a webworker for improved performance

Speaking of navigating back: Given that the new UI is based on a frontend framework including proper routing,
we now have a working back button. Amazing!

The only thing not yet available in the new UI is management of Zone and GoTo presets. This will be added later.
For now, please navigate to the old UI via the menu and use that.

#### Phone/Mobile
![image](https://user-images.githubusercontent.com/974410/138561830-e0a3edf7-1974-49cf-90a8-31e677b482d2.png) ![image](https://user-images.githubusercontent.com/974410/138561931-41b68344-f260-4343-9c79-ef85e56d9786.png)

![image](https://user-images.githubusercontent.com/974410/138561874-f5e5fee9-81dd-43fb-9de0-75263169a0e6.png) ![image](https://user-images.githubusercontent.com/974410/138561884-9633600b-3362-454b-b95d-90f8e5951971.png)

#### Tablet/Desktop

![image](https://user-images.githubusercontent.com/974410/138562037-05bc5140-d7af-488b-8734-72e66b820192.png)

![image](https://user-images.githubusercontent.com/974410/138561911-77aa8d10-3918-4eb7-96ff-8a6d0440dfce.png)

![image](https://user-images.githubusercontent.com/974410/138562111-3cbfe03c-7a19-4e57-9bfb-6b872239f432.png)

## Updater

Starting with this release, Valetudo will be capable of updating itself with the press of a button.

This has been a long-requested feature and considering how mature this whole project has become, I decided that
it was finally time to actually implement that.
As it is implemented now, it uses the GitHub API by default, meaning that I am not able to track you.

There's also no automatic update check, because periodical pings to some cloud service are obviously problematic.<br/>
You will decide when to press the "Check for Updates" button and also when to update.

## More stickers

I've decided to get a few more stickers printed because merch

![image](https://user-images.githubusercontent.com/974410/138592502-4c6d150e-ecde-46ea-9ce4-7da8fdb77431.png)

### Features

* **ui:** Add Valetudo logo to bottom sheet ([76eeaff](https://github.com/Hypfer/Valetudo/commit/76eeaffd4aa273e351d9306d4660df196380b275))
* **ui:** Bring back the old map renderer ([7aa6822](https://github.com/Hypfer/Valetudo/commit/7aa6822e20ef026a73bcfebcf76c9bbd49ce2a94))
* **ui:** Change map colors according to theme in use ([b17745e](https://github.com/Hypfer/Valetudo/commit/b17745e2ca70a9b24982f70cfc2272130cd1e9a4))
* **ui:** Fan speed icon and battery level bar improvements ([d0a0671](https://github.com/Hypfer/Valetudo/commit/d0a06712315f0dc7b93fc35ee9f9270a9204df7e))
* **ui:** Improve controls ([b678cda](https://github.com/Hypfer/Valetudo/commit/b678cdac0f2a15e1a3191f2125fbbaabbc7af833))
* **ui:** Map Editing ([35feb65](https://github.com/Hypfer/Valetudo/commit/35feb654e76599d0c267ed118453f3c25e4b8fbd))
* **ui:** Move locate to the map view ([32de2b8](https://github.com/Hypfer/Valetudo/commit/32de2b83daf1e03fa16b1115713a9290abc05029))
* **ui:** Place new client structures in the middle of the viewport ([d9d2a64](https://github.com/Hypfer/Valetudo/commit/d9d2a6436bd6ea24dd431b24776c93587114cfba))
* **ui:** Remove map transparency ([ace000b](https://github.com/Hypfer/Valetudo/commit/ace000b922a426076db49aaa25e1cbf4207b0ad8))
* **ui:** Remove obsolete segment controls ([9912cf5](https://github.com/Hypfer/Valetudo/commit/9912cf58f60a6ca88517786bb9fc6afccf9a5cf0))
* **updater:** Introduce ValetudoUpdaterNoUpdateRequiredState ([354028e](https://github.com/Hypfer/Valetudo/commit/354028e6f4f23e8c61a9f2e302edcaecf7807fe6))
* Updater ([5db2c55](https://github.com/Hypfer/Valetudo/commit/5db2c554dc9905352c310f4be7d6130a26f7730c))
* **ui:** Remove Accordion from interface settings ([c4c31bb](https://github.com/Hypfer/Valetudo/commit/c4c31bb5a68757a6ec687c3dde436f26374a2fdc))
* **ui:** Set the new UI as the default ([c0f90fb](https://github.com/Hypfer/Valetudo/commit/c0f90fb90f44d7020db016cfe9129d80f8df7d56))
* **ui:** Use a WebWorker for rendering the map layers when possible ([3eafded](https://github.com/Hypfer/Valetudo/commit/3eafded6c62a420795f6f910ca5b933f29a7e5f6))
* **ui:** Use real swagger icon in sidebar ([#1160](https://github.com/Hypfer/Valetudo/issues/1160)) ([f4c0093](https://github.com/Hypfer/Valetudo/commit/f4c0093527be1ae3ce6fa298bd878b3cc5b1283c))
* **ValetudoEvents:** Add Mop Attachment Reminder event ([ac268d7](https://github.com/Hypfer/Valetudo/commit/ac268d765f93c33aba14ad7ffd8602d00a02985e))
* Add nonce to every instance of a Valetudo map ([f0489f1](https://github.com/Hypfer/Valetudo/commit/f0489f15260f8eb368bc6b759029a022781bebf9))

### Bug Fixes

* Fix MopAttachmentReminderValetudoEvent not being a DismissibleValetudoEvent ([302f6ce](https://github.com/Hypfer/Valetudo/commit/302f6cefc3f155d45589ea1cf50ed39fc983b5d3))
* **mqtt:** Fix consumable state attribute data being published multiple times ([894bba9](https://github.com/Hypfer/Valetudo/commit/894bba9c9b99a5b67f584dbaa9e1571ffa18812c))
* **mqtt:** Fix zone cleanup ([#1130](https://github.com/Hypfer/Valetudo/issues/1130)) ([bb951b2](https://github.com/Hypfer/Valetudo/commit/bb951b2fcddb6435bfe6b838044cc2af7b0c235f))
* **mqtt:** Hopefully avoid a deadlock ([c3313ea](https://github.com/Hypfer/Valetudo/commit/c3313eac738cf8d5dc1a05f3c60308e4b002dd3a))
* **ui:** Allow more states for zone/segment cleanup and goto ([c738938](https://github.com/Hypfer/Valetudo/commit/c738938cc0d8140ce91011db8420df7a6768c893))
* **ui:** Better placement of segment labels ([ba788a0](https://github.com/Hypfer/Valetudo/commit/ba788a0da1042106cac698e5de9f20d6e7bb89f5))
* **ui:** Combinations of scrolling and touch events also should not cause any jank ([e36c40f](https://github.com/Hypfer/Valetudo/commit/e36c40f2a66a67baebd75da13ed8bb84cd26d371))
* **ui:** Fix controls jumping around when there is a status flag ([c641fb1](https://github.com/Hypfer/Valetudo/commit/c641fb10e3bc2a7e56a9e6a388be1d479fb6c91b))
* **ui:** Fix draw order for ControlsBottomSheet ([c46ef10](https://github.com/Hypfer/Valetudo/commit/c46ef10981298570bc14a7e9094698b70dd30e94))
* **ui:** Fix SSE not reconnecting after having closed the connection once ([6feaaf8](https://github.com/Hypfer/Valetudo/commit/6feaaf807065f3e5d585c9f723e8b48f9aa32621))
* **ui:** Further improve segment label placement using the median ([6545cc7](https://github.com/Hypfer/Valetudo/commit/6545cc7fb4077247b08675f8ea10895062996401))
* **ui:** Ignore map updates if the tab is invisible ([370753d](https://github.com/Hypfer/Valetudo/commit/370753dcd5bb2dd1803d8dac67d3b96759dd94c3))
* **ui:** MQTT port is an integer ([#1159](https://github.com/Hypfer/Valetudo/issues/1159)) ([5b91c96](https://github.com/Hypfer/Valetudo/commit/5b91c96eb8e2928be48b4ef63abbb7c043b34aa9))
* **ui:** Postpone map data updates during pan/zoom to prevent jank ([92a9147](https://github.com/Hypfer/Valetudo/commit/92a9147191981b15d48e4817f7200875470a13fe))
* **ui:** Postpone map data updates during scroll to prevent jank ([adcd117](https://github.com/Hypfer/Valetudo/commit/adcd1174a50b4bce4805da5a36bcf80646eb9979))
* **ui:** Prevent issues with zooming on mobile ([a5057fd](https://github.com/Hypfer/Valetudo/commit/a5057fd7f4dd58b38628f5bbdd4ade951ca6b6fc))
* **ui:** Use the correct canvas size for the map layer renderer ([b55ffa6](https://github.com/Hypfer/Valetudo/commit/b55ffa6cc56d7c7d6bb0c8459ccd1d35252c3b63))
* **ui:** Zone dimensions should be based on the pixelSize ([b317c85](https://github.com/Hypfer/Valetudo/commit/b317c856dbdfe95ab45ea639223d23fca093c464))
* **updater:** Fix fallback download location for roborock ([d515f04](https://github.com/Hypfer/Valetudo/commit/d515f04e5f6695b71567f6818ba2734553c75d03))
* **vendor.dreame:** Fix virtual restriction coordinates being not sorted properly ([17cc303](https://github.com/Hypfer/Valetudo/commit/17cc303cf4726cf4bcbb9597c268273896b52c78))
* **vendor.dreame:** Ignore error 68 ([dd44320](https://github.com/Hypfer/Valetudo/commit/dd44320a390c42f6e36a301931e5006e103a4684))
* **webserver:** Allow unsetting names of segments ([f992e5a](https://github.com/Hypfer/Valetudo/commit/f992e5adce4c4fd997ff4a6c868f95d32a5e214b))
* Use shorter model names ([d268013](https://github.com/Hypfer/Valetudo/commit/d268013b9c93b4dc610cb8374a3316d9e2a842a9))

<hr /><em>This discussion was created from the release <a href='https://github.com/Hypfer/Valetudo/releases/tag/2021.11.0'>Valetudo 2021.11.0</a>.</em>

**Comment by LayerCakeMakes** [comment]
Where can we buy the stickers/merch?

**Reply by Hypfer** [reply]
That's a very good question. Honestly, I didn't think about that just yet. 😅

Any suggestions?

**Reply by LayerCakeMakes** [reply]
Hmm I can only think of ebay or setting up a shopify site.

**Reply by marcelfischer** [reply]
I would like to buy a few ;)

**Reply by lukaswimmer** [reply]
I do not know if you are keen on sending a lot of letters @Hypfer. When distributing it from a central place is too much work, something like a phone chain could work. @Hypfer sends a pack to me, I take two stickers out, send it to the next one, etc. But it relies heavily on trust that everybody forwards the stickers. And maybe it only works within every country separately.

**Reply by Hypfer** [reply]
I've just created a redbubble store here: https://www.redbubble.com/people/Valetudo/shop

It was surprisingly easy requiring no monthly fees or any other work.
The downside of that is that the prices are higher than one would expect for a few stickers.

Still, this option now exists for people that just want stickers without printing them themselves.
There is the option to do that though. The SVGs can be found in the assets folder of the repo

**Reply by LayerCakeMakes** [reply]
@Hypfer I bought a few have to say the quality is not great pretty grainy and the cutout is not centered or straight.

**Reply by Hypfer** [reply]
Oh no :(
I'm sorry to hear that. I thought that given the size of redbubble that wouldn't be an issue.
The account there has been deleted so that this won't happen again.

If you send me your address via my email address found on my profile page, I can send you some of the ones I have here.
Free of charge of course.

**Reply by LayerCakeMakes** [reply]
Send the mail, thank you for the offer. It seems redbubble only does management the printing is external. The image itself might have been fine if not looked at very closely (though I did order the smallest size option larger ones might be a problem) but the frameing is just strange. Tell me if I should send you the stickers then you can judge the quality yourself.

**Reply by lukaswimmer** [reply]
![3741AE0A-46E4-41B0-935E-764BD2584EDA](https://user-images.githubusercontent.com/6631057/153628751-9b9a084e-9cae-46de-aab4-00db58de368f.jpeg)

I ordered the small ones without border and I am happy. Are these ones different from yours @kbehren?

**Reply by LayerCakeMakes** [reply]
Yeah you seem to have a mostly consistent border mine is really offset 2 mm left and top, 4mm right and bottom and the rounded corners do not follow the curve of the logo on mine. If I get a light so that the cut lines show up I'll take a picture.

**Reply by LayerCakeMakes** [reply]
This is the best I could do so the cutlines show up.
![Sticker](https://user-images.githubusercontent.com/1534212/153635816-5b652f24-be18-4c51-961c-3512c79f6300.jpg)
